File: dragonsquad/craneam/com/UIElement.java
package dragon.craneam.com;

public class UIElement
  extends Quad
  {
    private static final String TAG = "Dragon Squad";  // Game tag in logs
    private static final boolean DEBUG = false;    // Debug flag
    private int width;
    private int height;
    private int xPosition;
    private int yPosition;

    public UIElement(int w, int h)
      {
        super(w, h);
        width = w;
        height = h;
        textureWidth = 256;
        textureHeight = 256;
      }

    public int GetWidth()
      {
        return width;
      }

    public int GetHeight()
      {
        return height;
      }

    public int GetX()
      {
        return xPosition;
      }

    public int GetY()
      {
        return yPosition;
      }

    public void SetXPosition(int value)
      {
        for (int i = 0; i < 12; i += 3)
          {
            vertices[i] += value;
          }
        xPosition = value;
      }

    public void SetYPosition(int value)
      {
        for (int i = 1; i < 12; i += 3)
          {
            vertices[i] += value;
          }
        yPosition = value;
      }

    public void SetTextureLocation(int id)
      {
        switch (id)
          {
          case 0:
            SetTextureCoordinates(0, 255, 0, 255);
            break;
          }
      }
  }

File: app/src/main/java/com/koodipuukko/dragonsquad/UIElement.kt
package com.koodipuukko.dragonsquad

class UIElement
	(private val width: Int, private val height: Int) : Quad(width, height) {
	private var xPosition = 0
	private var yPosition = 0

	init {
		textureWidth = 256
		textureHeight = 256
	}

	fun getWidth(): Int {
		return width
	}

	fun getHeight(): Int {
		return height
	}

	fun getX(): Int {
		return xPosition
	}

	fun getY(): Int {
		return yPosition
	}

	fun setXPosition(value: Int) {
		var i = 0
		while(i < 12) {
			vertices[i] = vertices[i] + value
			i += 3
		}
		xPosition = value
	}

	fun setYPosition(value: Int) {
		var i = 1
		while(i < 12) {
			vertices[i] = vertices[i] + value
			i += 3
		}
		yPosition = value
	}

	fun setTextureLocation(id: Int) {
		when (id) {
			0 -> setTextureCoordinates(0, 255, 0, 255)
		}
	}

	companion object {
		private const val TAG = "Dragon Squad" // Game tag in logs
		private const val DEBUG = false // Debug flag
	}
}
